xml/value.go

```diff
@@ -0,0 +1,339 @@
+package xml
+
+import (
+	"encoding/base64"
+	"fmt"
+	"math"
+	"math/big"
+	"strconv"
+	"strings"
+)
+
+// Value represents an XML Value type
+// XML Value types: Object, Array, Map, String, Number, Boolean, and Null.
+type Value struct {
+	w       writer
+	scratch *[]byte
+
+	startElement *StartElement
+	endElement   *EndElement
+}
+
+// newValue returns a new Value encoder
+func newValue(w writer, scratch *[]byte, startElement *StartElement, endElement *EndElement) Value {
+	return Value{
+		w:            w,
+		scratch:      scratch,
+		startElement: startElement,
+		endElement:   endElement,
+	}
+}
+
+// writeStartElement takes in a start element and writes it.
+// It handles namespace, attributes in start element.
+func writeStartElement(w writer, el *StartElement) error {
+	if el == nil {
+		return nil
+	}
+
```

Comment:
could you add a comment why `el` might be nil? I'd imagine it would be an error in code path if `el` were nil unexpectedly.